### ISSUE TITLE
feat: allow to register plugin as feature flag

### ISSUE THREAD

**Issue by accezar** [drodil/backstage-plugin-qeta#24]
Offer possibility to register the plugin as feature flag. 
Ex: `<FeatureFlagged with="feature-flag-queta-plugin">`

```
export const qetaPlugin = createPlugin({
  id: 'qeta',
  featureFlags: [{ name: 'feature-flag-queta-plugin' }],
 // ...
});
```

https://github.com/drodil/backstage-plugin-qeta/blob/4ab4ddd4f3f70a3a1fcb09d3ca29afcb3e06d497/plugins/qeta/src/plugin.ts#L12

**Comment by drodil** [comment]
Hi @accezar ,

Would you use this feature to hide the whole plugin behind a feature flag or what is your idea? Because I had the exact same thing in my mind previously but decided to make a new backstage feature that allows enabling feature flags in the application instead of the plugin and then those can be used to hide, for example, menu items in the frontend. See details here: https://github.com/backstage/backstage/pull/15556 

The application level feature flags will be available in the next backstage release 1.10.0 on 17th of January.

**Comment by accezar** [comment]
@drodil I was thinking about hide the whole plugin, but application level looks promising!

**Comment by drodil** [comment]
Yes that's exactly what you can do with the application level feature flags! I'll close this issue as it will can be handled with that feature.

**Comment by drodil** [comment]
@accezar backstage version 1.10.0 was just released so if you want to hide the plugin behind feature flag you can follow these instructions: https://backstage.io/docs/plugins/feature-flags
